inspirehep/modules/orcid/tasks.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+import os
+
+import sys
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+
+from flask import current_app
+
+from invenio_db import db
+
+from invenio_pidstore.resolver import Resolver
+
+from invenio_search import current_search_client
+from invenio_search.utils import schema_to_index
+
+from requests import RequestException
+
+from .models import InspireOrcidRecords
+from .utils import convert_to_orcid, get_authors_credentials
+
+logger = get_task_logger(__name__)
+
+
+@shared_task(ignore_result=True)
+def send_to_orcid(sender):
+    """Sends records to orcid."""
+    index, doc_type = schema_to_index(sender['$schema'])
+    # FIXME: next two lines are there for debugging reasons.
+    if 'authors' in sender:
+        sender['authors'].append({u'affiliations': [{u'value': u'St. Petersburg, INP'}],  u'curated_relation': True,  u'full_name': u'Paparrigopoulos, Panos',  u'profile': {
+                                 u'__url__': u'http://inspirehep.net/record/1064588'},  u'record': {u'$ref': u'http://localhost:5000/api/authors/10'}})
+    if doc_type == 'hep':
+        logger.info("Sending " + sender.get('control_number') + " to orcid.")
+        try:
+            api = current_app.extensions['inspire-orcid'].orcid_api
```

Comment:
For testability `api` should be injected in the method. Wouldn't hurt to call it `orcid_api`, since it's what it is :D